jfrog-cli/artifactory/utils/golang/project/dependencies/dependenciesutils.go

```diff
@@ -0,0 +1,452 @@
+func createDependency(cachePath, dependencyName, version string) (*Package, error) {
+	// We first check if the this dependency has a zip binary in the local go cache.
+	// If it does not, nil is returned. This seems to be a bug in go.
+	zipPath, err := getPackageZipLocation(cachePath, dependencyName, version)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if zipPath == "" {
+		return nil, nil
+	}
+
+	dep := Package{}
+
+	dep.id = strings.Join([]string{dependencyName, version}, ":")
+	dep.version = version
+	dep.zipPath = zipPath
+	dep.modContent, err = ioutil.ReadFile(filepath.Join(cachePath, dependencyName, "@v", version+".mod"))
+	if err != nil {
+		return &dep, errorutils.CheckError(err)
+	}
+
+	// Mod file dependency
+	modDependency := buildinfo.Dependency{Id: dep.id}
+	checksums, err := checksum.Calc(bytes.NewBuffer(dep.modContent))
+	if err != nil {
+		return &dep, err
+	}
+	modDependency.Checksum = &buildinfo.Checksum{Sha1: checksums[checksum.SHA1], Md5: checksums[checksum.MD5]}
+
+	// Zip file dependency
+	zipDependency := buildinfo.Dependency{Id: dep.id}
+	fileDetails, err := fileutils.GetFileDetails(dep.zipPath)
+	if err != nil {
+		return &dep, err
+	}
+	zipDependency.Checksum = &buildinfo.Checksum{Sha1: fileDetails.Checksum.Sha1, Md5: fileDetails.Checksum.Md5}
+
+	dep.buildInfoDependencies = append(dep.buildInfoDependencies, modDependency, zipDependency)
+	return &dep, nil
+}
+
+// Returns the path to the package zip file if exists.
+func getPackageZipLocation(cachePath, dependencyName, version string) (string, error) {
+	zipPath, err := getPackagePathIfExists(cachePath, dependencyName, version)
+	if err != nil {
+		return "", err
+	}
+
+	if zipPath != "" {
+		return zipPath, nil
+	}
+
+	zipPath, err = getPackagePathIfExists(filepath.Dir(cachePath), dependencyName, version)
+
+	if err != nil {
+		return "", err
+	}
+
+	return zipPath, nil
+}
+
+// Validates if the package zip file exists.
+func getPackagePathIfExists(cachePath, dependencyName, version string) (zipPath string, err error) {
+	zipPath = filepath.Join(cachePath, dependencyName, "@v", version+".zip")
+	fileExists, err := fileutils.IsFileExists(zipPath, false)
+	if err != nil {
+		log.Warn(fmt.Sprintf("Could not find zip binary for dependency '%s' at %s.", dependencyName, zipPath))
+		return "", err
+	}
+	// Zip binary does not exist, so we skip it by returning a nil dependency.
+	if !fileExists {
+		log.Debug("The following file is missing:", zipPath)
+		return "", nil
+	}
+	return zipPath, nil
+}
+
+func getGOPATH() (string, error) {
+	goCmd, err := golang.NewCmd()
+	if err != nil {
+		return "", err
+	}
+	goCmd.Command = []string{"env", "GOPATH"}
+	output, err := utils.RunCmdOutput(goCmd)
+	if err != nil {
+		return "", fmt.Errorf("Could not find GOPATH env: %s", err.Error())
+	}
+	return strings.TrimSpace(string(output)), nil
+}
+
+func mergeReplaceDependenciesWithGraphDependencies(replaceDeps []string, graphDependencies map[string]bool) error {
```

Comment:
The function's argument name seem to have inconsistent format.
I suggest renaming *graphDependencies* to *graphDeps*.
Also, this function never seems to return an error, so we can remove from the signature the return value.

Reply:
Done